app/controllers/AddressController.scala

```diff
@@ -712,57 +702,60 @@ class AddressController @Inject()(
 
   private def submitConfirmClosePostalAddress(nino: Nino, personDetails: PersonDetails)(
     implicit request: UserRequest[_]): Future[Result] = {
-    def internalServerError =
-      InternalServerError(
-        error(
-          "global.error.InternalServerError500.title",
-          Some("global.error.InternalServerError500.title"),
-          List("global.error.InternalServerError500.message")
-        ))
 
     val address = getAddress(personDetails.correspondenceAddress)
     val closingAddress = address.copy(endDate = Some(LocalDate.now), startDate = Some(LocalDate.now))
 
-    for {
-      response <- citizenDetailsService.updateAddress(nino, personDetails.etag, closingAddress)
-      action <- response match {
-                 case UpdateAddressBadRequestResponse =>
-                   Future.successful(
-                     BadRequest(
-                       error(
-                         "global.error.BadRequest.title",
-                         Some("global.error.BadRequest.title"),
-                         List("global.error.BadRequest.message1", "global.error.BadRequest.message2")
-                       )
-                     )
-                   )
-                 case UpdateAddressUnexpectedResponse(_) | UpdateAddressErrorResponse(_) =>
-                   Future.successful(internalServerError)
-                 case UpdateAddressSuccessResponse =>
-                   for {
-                     _ <- auditConnector.sendEvent(
-                           buildEvent(
-                             "closedAddressSubmitted",
-                             "closure_of_correspondence",
-                             auditForClosingPostalAddress(closingAddress, personDetails.etag, "correspondence")))
-                     _        <- clearCache() //This clears ENTIRE session cache, no way to target individual keys
-                     inserted <- editAddressLockRepository.insert(nino.withoutSuffix, PostalAddrType)
-                     _        <- addressMovedService.moved(address.postcode.getOrElse(""), address.postcode.getOrElse(""))
-                   } yield
-                     if (inserted) {
-                       Ok(
-                         updateAddressConfirmationView(
-                           PostalAddrType,
-                           closedPostalAddress = true,
-                           Some(getAddress(personDetails.address).fullAddress),
-                           None
+    citizenDetailsService.getEtag(nino.nino) flatMap { etag =>
+      etag match {
+        case None =>
+          Logger.warn("Failed to retrieve Etag from citizen-details")
```

Comment:
Should probably log these as errors, if we have no ETag the user can't do anything anyway